strongbox-storage/strongbox-storage-core/src/main/java/org/carlspring/strongbox/repository/AbstractRepositoryManagementStrategy.java

```diff
@@ -94,4 +96,25 @@ public Configuration getConfiguration()
         return configurationManagementService.getConfiguration();
     }
 
+    public static boolean shouldDownloadAllRemoteRepositoryIndexes()
```

Comment:
These methods are specific to Maven only. This is why I placed them in the `MavenRepositoryManagementStrategy`. The NuGet repositories don't have a Maven Index, so this isn't really applicable to them and should be restored as it was. The actual check should be in the `NugetRepositoryManagementStrategy` class.

I remember that the other day you mentioned that this should better be done using Spring profiles. I am not quite sure how this help (but please feel free to elaborate). My impression is that either way, (whether using Spring Profiles, or the properties passed in to the `maven-surefire-plugin` and `maven-failsafe-plugin`), this has the same effect. I admit, I am not a Spring guru like you guys, so maybe I'm misunderstanding something, so please feel free to comment. :)

Reply:
@carlspring I understand that it looks crooked, but to have many such configuration properties will be not less crooked.
You will need not to forget to put them in maven everywhere you need them. I suggest to create a task to use Spring profiles to disable Cron tasks within UnitTests and implement this task instead of all of this system properties configuration.

Reply:
The thing is -- I've been trying really hard to keep Maven and NuGet concepts separated so that there is no unnecessary confusion. I had finally split things apart with the configuration file changes the other day... I'd really like to keep things separate.